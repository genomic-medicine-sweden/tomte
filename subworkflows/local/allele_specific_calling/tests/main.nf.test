nextflow_workflow {

    name "Test Workflow ALLELE_SPECIFIC_CALLING"
    script "../main.nf"
    workflow "ALLELE_SPECIFIC_CALLING"
    config "./nextflow.config"
    tag "allele_specific_calling"

    setup {
        run("GUNZIP", alias: "GUNZIP_FASTA") {
            script "../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'fasta' ],
                    file("${projectDir}/test_data/grch37_chr21.fa.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("SAMTOOLS_FAIDX") {
            script "../../../../modules/nf-core/samtools/faidx/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip
                input[1] = [[],[]]
                """
            }
        }

        run("GATK4_CREATESEQUENCEDICTIONARY", alias: "BUILD_DICT") {
            script "../../../../modules/nf-core/gatk4/createsequencedictionary/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip
                """
            }
        }

        run("SAMTOOLS_CONVERT", alias: "CRAM_TO_BAM") {
            script "../../../../modules/nf-core/samtools/convert/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ case:'finequagga', id:'ACC5963A3', sample:'ACC5963A3', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:false ],
                    file("${projectDir}/test_data/ACC5963A3.cram", checkIfExists: true),
                    file("${projectDir}/test_data/ACC5963A3.cram.crai", checkIfExists: true)
                    ])
                input[1] = GUNZIP_FASTA.out.gunzip
                input[2] = SAMTOOLS_FAIDX.out.fai
                """
            }
        }

    }

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                    input[0] = Channel.of([
                        [ case:'finequagga', id:'ACC5963A1', sample:'ACC5963A1', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:false ],
                        file("${projectDir}/test_data/ACC5963A1_split_rmdup_info.vcf.gz", checkIfExists: true),
                        file("${projectDir}/test_data/ACC5963A1_split_rmdup_info.vcf.gz.tbi", checkIfExists: true)
                        ])
                    input[1] = Channel.of([
                        [ case:'finequagga', id:'ACC5963A1', sample:'ACC5963A1', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:false ],
                        file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                        file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                        ])
                    input[2] = GUNZIP_FASTA.out.gunzip.collect()
                    input[3] = SAMTOOLS_FAIDX.out.fai.collect()
                    input[4] = BUILD_DICT.out.dict.collect()
                    input[5] = Channel.of([probands:['ACC5963A1'], id:'finequagga']).collect()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcf.collect{path(it[1]).vcf.getVariantsAsStrings(2)},
                    file(workflow.out.tbi.get(0).get(1)).name,
                    workflow.out.versions,
                ).match()}
            )
        }

    }

    test("Should run with two samples") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                ch_cram_converted = CRAM_TO_BAM.out.bam.collect().join(CRAM_TO_BAM.out.bai.collect())
                // Channel for the BAM sample
                ch_bam = Channel.of([
                    [ case:'finequagga', id:'ACC5963A1', sample:'ACC5963A1', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:false ],
                    file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                    file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                    ])
                ch_bam_files = ch_bam.mix(ch_cram_converted)

                input[0] = Channel.of([
                        [ case:'finequagga', id:'ACC5963A1', sample:'ACC5963A1', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:false ],
                        file("${projectDir}/test_data/ACC5963A1_split_rmdup_info.vcf.gz", checkIfExists: true),
                        file("${projectDir}/test_data/ACC5963A1_split_rmdup_info.vcf.gz.tbi", checkIfExists: true)
                        ],
                        [[ case:'finequagga', id:'ACC5963A3', sample:'ACC5963A3', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:false ],
                        file("${projectDir}/test_data/ACC5963A3_split_rmdup_info.vcf.gz", checkIfExists: true),
                        file("${projectDir}/test_data/ACC5963A3_split_rmdup_info.vcf.gz.tbi", checkIfExists: true)
                        ])
                input[1] = ch_bam_files
                input[2] = GUNZIP_FASTA.out.gunzip.collect()
                input[3] = SAMTOOLS_FAIDX.out.fai.collect()
                input[4] = BUILD_DICT.out.dict.collect()
                input[5] = Channel.of([probands:['ACC5963A1','ACC5963A3'], id:'finequagga']).collect()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcf.collect{path(it[1]).vcf.getVariantsAsStrings(2)},
                    file(workflow.out.tbi.get(0).get(1)).name,
                    workflow.out.versions,
                ).match()}
            )
        }

    }

}
