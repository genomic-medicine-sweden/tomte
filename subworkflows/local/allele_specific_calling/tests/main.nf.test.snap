{
    "Should run with two samples": {
        "content": [
            [
                "VcfFile [chromosomes=[chr21], sampleCount=2, variantCount=805, phased=false, phasedAutodetect=false]"
            ],
            [
                [
                    "chr21\t9656524\t.\tA\tG\t7.30814\t.\tVDB=0.18;SGB=-0.453602;MQ0F=0;MQ=20;FOUND_IN=bcftools;DP=6;DP4=0,0,4,0;AN=4;AC=4\tGT:PL\t1/1:36,6,0\t1/1:36,6,0",
                    "chr21\t9656547\t.\tG\tA\t7.30814\t.\tVDB=0.22;SGB=-0.453602;MQ0F=0;MQ=20;FOUND_IN=bcftools;DP=8;DP4=0,0,4,0;AN=4;AC=4\tGT:PL\t1/1:36,6,0\t1/1:36,6,0"
                ]
            ],
            "finequagga_split_rmdup.vcf.gz.tbi",
            [
                "versions.yml:md5,4260f81272bc7976004cb71c220ac7c4",
                "versions.yml:md5,53dc76802ccd9d3604c413cea5810d43",
                "versions.yml:md5,938667310a4b39815b7d6c59e90be868",
                "versions.yml:md5,a0fd1d34c375ce6de3c12f7677455568",
                "versions.yml:md5,a265f17fbd2eeb9eda1f4e876bc1ebcc",
                "versions.yml:md5,d2143d7224a883879775c66b56ca2912",
                "versions.yml:md5,dfa95053efa28dd1e5be7fa129fe83b9",
                "versions.yml:md5,f375509729cb99d7a6f8073446f84f00",
                "versions.yml:md5,ff29ff5a2e422142521dd23401bb87f8"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.5"
        },
        "timestamp": "2025-07-29T15:37:21.545753406"
    },
    "Should run without failures": {
        "content": [
            [
                "VcfFile [chromosomes=[chr21], sampleCount=1, variantCount=805, phased=false, phasedAutodetect=false]"
            ],
            [
                [
                    "chr21\t9656524\t.\tA\tG\t7.30814\t.\tDP=3;VDB=0.18;SGB=-0.453602;MQ0F=0;AC=2;AN=2;DP4=0,0,2,0;MQ=20;FOUND_IN=bcftools\tGT:PL\t1/1:36,6,0",
                    "chr21\t9656547\t.\tG\tA\t7.30814\t.\tDP=4;VDB=0.22;SGB=-0.453602;MQ0F=0;AC=2;AN=2;DP4=0,0,2,0;MQ=20;FOUND_IN=bcftools\tGT:PL\t1/1:36,6,0"
                ]
            ],
            "finequagga_split_rmdup.vcf.gz.tbi",
            [
                "versions.yml:md5,00ddd8c72d45720f971db98815fbfc0e",
                "versions.yml:md5,4260f81272bc7976004cb71c220ac7c4",
                "versions.yml:md5,53dc76802ccd9d3604c413cea5810d43",
                "versions.yml:md5,938667310a4b39815b7d6c59e90be868",
                "versions.yml:md5,a0fd1d34c375ce6de3c12f7677455568",
                "versions.yml:md5,a265f17fbd2eeb9eda1f4e876bc1ebcc",
                "versions.yml:md5,d2143d7224a883879775c66b56ca2912",
                "versions.yml:md5,f375509729cb99d7a6f8073446f84f00",
                "versions.yml:md5,ff29ff5a2e422142521dd23401bb87f8"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.5"
        },
        "timestamp": "2025-07-29T15:36:43.313979082"
    }
}